agent/agent.go

```diff
@@ -405,3 +364,32 @@ func (a *Agent) StartStatsd() {
 		}
 	}()
 }
+
+func connectWithBackoff(url string, tlsOpts *types.TLSOptions, header http.Header) transport.Transport {
+	var conn transport.Transport
+
+	backoff := retry.ExponentialBackoff{
+		InitialDelayInterval: 500 * time.Millisecond,
+		MaxDelayInterval:     10 * time.Second,
+		MaxRetryAttempts:     0, // Unlimited attempts
+		Multiplier:           1.5,
+	}
```

Comment:
Should these be exposed as part of the agent configuration?

Reply:
I'd rather not let the scope of this PR creep any more. Can we file an issue instead of pulling this in?